bubblewrap.c

```diff
@@ -2162,7 +2168,7 @@ main (int    argc,
 
   __debug__ (("forking for child\n"));
 
-  if (opt_unshare_pid || lock_files != NULL || opt_sync_fd != -1)
+  if (!no_reaper && (opt_unshare_pid || lock_files != NULL || opt_sync_fd != -1))
```

Comment:
Also, if opt_unshare_pid is set *and no_reaper, then we should not allocate event_fd (because it needs the init process).